src/llm/wmt_model.py

```diff
@@ -247,20 +270,74 @@ def __init__(
                     model_dim=model_dim,
                     num_heads=num_heads,
                     expansion_dim=expansion_dim,
+                    dropout_rate=dropout_rate,
                 )
             )
 
     def forward(
         self,
-        decoder_input: Float[Tensor, "batch dec_seq_len model_dim"],
-        encoder_outputs: List[Float[Tensor, "batch enc_seq_len model_dim"]],
-    ) -> Float[Tensor, "batch dec_seq_len model_dim"]:
+        decoder_input: Float[Tensor, "batch dec_seq_len model_dim"],  # type: ignore
+        encoder_outputs: List[Float[Tensor, "batch enc_seq_len model_dim"]],  # type: ignore
+    ) -> Float[Tensor, "batch dec_seq_len model_dim"]:  # type: ignore
         x = decoder_input
         for idx, decoder_layer in enumerate(self.decoder_layers):
             x = decoder_layer(decoder_input=x, encoder_outputs=encoder_outputs[idx])
         return x
 
 
+class Embedding(nn.Module):
+    def __init__(self, model_dim: int, vocab_size: int):
+        super().__init__()
+        self.embedding = nn.Embedding(vocab_size, model_dim)
+        self.scale = math.sqrt(model_dim)
+
+    def forward(self, tokens: Float[Tensor, "batch seq_len"]) -> Float[Tensor, "batch seq_len model_dim"]:  # type: ignore
+        embedding = self.embedding(tokens) * self.scale
```

Comment:
Can you point to the place where we need to scale the embeddings? This is really interesting (and can form an ablation later)

Reply:
from the paper, 

"In the embedding layers, we multiply those weights by √dmodel."

<img width="1159" alt="image" src="https://github.com/user-attachments/assets/5a38e1f0-1bf5-4edf-995a-7a06856f0865" />

Reply:
Can we comment on top of the scaling bit "as suggested in the paper (arxiv link) in section 3.4"